controllers/cohortdata.go

```diff
@@ -44,6 +49,15 @@ func (u CohortDataController) RetrieveHistogramForCohortIdAndConceptId(c *gin.Co
 	cohortId, _ := strconv.Atoi(cohortIdStr)
 	histogramConceptId, _ := strconv.ParseInt(histogramIdStr, 10, 64)
 
+	uniqueCohortDefinitionIdsList := utils.GetUniqueCohortDefinitionIdsListFromRequest([]int{cohortId}, cohortPairs)
```

Comment:
Is it a lof of effort to write this logic block into a function? Since it's also reused at 102-110, 256-264

Reply:
it would only results in these lines being moved there:
```
		log.Printf("Error: invalid request")
		c.JSON(http.StatusBadRequest, gin.H{"message": "access denied"})
		c.Abort()
```
I think it is not worth it and potentially results in less clear code, especially given the rest of the code around it following a similar structure.

Reply:
I was thinking we could have something like this:
```
func (u TeamProjectAuthz) ValidateTeamProject (c *gin.Context, cohortIds []int, []CustomDichotomousVariableDef) bool {

**paste in block of code here**

}
```

Reply:
it would still require something like this at the outer layer:
```
	validAccessRequest := ValidateTeamProject( .... )
	if !validAccessRequest {
		return
	}
```
so I think the gain is minimal... It does make part of it easier to test though, so that might make it worth it... 🤔  
Let me add a TODO

Reply:
Ok, I made a small improvement in the code that improves reusability of the existing `TeamProjectValidation` method. I think this gets the code closer to what you mentioned.